core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala

```diff
@@ -202,12 +202,13 @@ class JobProgressListener(conf: SparkConf) extends SparkListener with Logging {
     }
     jobIdToData(jobStart.jobId) = jobData
     activeJobs(jobStart.jobId) = jobData
-    for (stageId <- jobStart.stageIds) {
-      stageIdToActiveJobIds.getOrElseUpdate(stageId, new HashSet[StageId]).add(jobStart.jobId)
-    }
+
     // If there's no information for a stage, store the StageInfo received from the scheduler
     // so that we can display stage descriptions for pending stages:
     for (stageInfo <- jobStart.stageInfos) {
+      stageIdToActiveJobIds.getOrElseUpdate(stageInfo.stageId, 
```

Comment:
It's not immediately clear to me they're equivalent. They'd have to contain the same set of stage IDs. Is it worth it?